qemu/tests/netperf_udp_perf.py

```diff
@@ -0,0 +1,229 @@
+                               username, shell_port, password)
+
+    netperf_base.tweak_tuned_profile(params, server_ctl, client, host)
+
+    env.stop_ip_sniffing()
+
+    try:
+        error_context.context("Start netperf udp stream testing", logging.info)
+        start_test(server_ip, server_ctl, host, client, test.resultsdir,
+                   test_duration=int(params.get('l')),
+                   burst_time=params.get('burst_time'),
+                   numbers_per_burst=params.get('numbers_per_burst'),
+                   ver_cmd=params.get('ver_cmd', "rpm -q qemu-kvm"),
+                   params=params, test=test)
+
+        if params.get("log_hostinfo_script"):
+            src = os.path.join(test.virtdir, params.get("log_hostinfo_script"))
+            path = os.path.join(test.resultsdir, "systeminfo")
+            process.system_output("bash %s %s &> %s" % (
+                                  src, test.resultsdir, path), shell=True)
+
+        if params.get("log_guestinfo_script") and params.get("log_guestinfo_exec"):
+            src = os.path.join(test.virtdir, params.get("log_guestinfo_script"))
+            path = os.path.join(test.resultsdir, "systeminfo")
+            destpath = params.get("log_guestinfo_path", "/tmp/log_guestinfo.sh")
+            vm.copy_files_to(src, destpath, nic_index=1)
+            logexec = params.get("log_guestinfo_exec", "bash")
+            output = server_ctl.cmd_output("%s %s" % (logexec, destpath))
+            logfile = open(path, "a+")
+            logfile.write(output)
+            logfile.close()
+    except process.CmdError as err:
+        test.cancel("test faild")
+
+
+@error_context.context_aware
+def start_test(server, server_ctl, host, client, resultsdir,
+               test_duration="20",
+               burst_time="1",
+               numbers_per_burst="1000 1500 2000 2500 3000",
+               ver_cmd=None,
+               params=None, server_cyg=None, test=None):
+    """
+    Start to test with different combination of burst_time and numbers_per_burst
+
+    """
+    if params is None:
+        params = {}
+
+    guest_ver_cmd = params.get("guest_ver_cmd", "uname -r")
+    fd = open("%s/netperf-udp-perf.result.%s.RHS" % (
+              resultsdir, time.time()), "w")
+
+    test.write_test_keyval({'kvm-userspace-ver': netperf_base.ssh_cmd(
+                            host, ver_cmd).strip()})
+    test.write_test_keyval({'guest-kernel-ver': netperf_base.ssh_cmd(
+                            server_ctl, guest_ver_cmd).strip()})
+    test.write_test_keyval({'session-length': test_duration})
+
+    fd.write('### kvm-userspace-ver : %s\n' % netperf_base.ssh_cmd(
+                                            host, ver_cmd).strip())
+    fd.write('### guest-kernel-ver : %s\n' % netperf_base.ssh_cmd(
+                                server_ctl, guest_ver_cmd).strip())
+    fd.write('### kvm_version : %s\n' % os.uname()[2])
+    fd.write('### session-length : %s\n' % test_duration)
```

Comment:
The above part seems also duplicated with netperf.py, would you consider abstract a function for it?

Reply:
update.